kt/godot-library/src/main/kotlin/godot/runtime/Bootstrap.kt

```diff
@@ -36,67 +26,18 @@ internal class Bootstrap {
             doInitGraal()
         } else {
             val userCodeFile = File(userCodePath)
-
-            if (userCodeFile.exists()) {
-                doInit(userCodeFile.toURI().toURL(), loader)
-            } else {
-                if (projectRootDir.isNotEmpty()) {
-                    JVMLogging::warning
-                } else {
-                    JVMLogging::error
-                }.invoke("No main.jar detected at $userCodeFile. No script will be loaded. Build the gradle project to load scripts.")
-            }
-
-            if (projectRootDir.isNotEmpty()) {
-                watchService = FileSystems.getDefault().newWatchService()
-                val watchKey = getBuildLockDir(projectRootDir).toPath().register(
-                    watchService,
-                    StandardWatchEventKinds.ENTRY_CREATE,
-                    StandardWatchEventKinds.ENTRY_DELETE,
-                    StandardWatchEventKinds.ENTRY_MODIFY,
-                )
-
-                executor = Executors.newSingleThreadScheduledExecutor { runnable ->
-                    val thread = Thread(runnable)
-                    thread.isDaemon = true
-                    thread
-                }
-
-                executor!!.scheduleAtFixedRate({
-                    val events = watchKey.pollEvents()
-                    if (events.isNotEmpty()) {
-                        if (File(getBuildLockDir(projectRootDir), "buildLock.lock").exists()) {
-                            JVMLogging.info("Build lock present. Not reloading...")
-                            return@scheduleAtFixedRate
-                        }
-                        JVMLogging.info("Changes detected, reloading classes ...")
-
-                        if (::serviceLoader.isInitialized) {
-                            clearClassesCache()
-                            serviceLoader.reload()
-                        }
-
-                        if (userCodeFile.exists()) {
-                            doInit(userCodeFile.toURI().toURL(), null) //no classloader so new main jar gets loaded
-                        } else {
-                            JVMLogging.warning("No main.jar detected. No classes will be loaded. Build the project to load classes")
-                        }
-                    }
-                }, 3, 3, TimeUnit.SECONDS)
-            }
+            doInit(userCodeFile.toURI().toURL(), loader)
         }
     }
 
     fun finish() {
-        executor?.shutdown()
-        watchService?.close()
+        Thread.currentThread().contextClassLoader = null
```

Comment:
Why do we set the context classloader to null here explicitly?

Reply:
It was already there before. We first set it from the C++ side, but somehow remove it from the Kotlin side.